sysbox-in-docker/README.md

```diff
@@ -0,0 +1,112 @@
+[doc](../docs/distro-compat.md).
+
+* Root-privileges: As a container runtime, Sysbox requires root privileges to
+operate. As a result, the Sysbox-In-Docker container must be launched in
+"privileged" mode.
+
+    **Note**: Within the privileged container, inner containers launched with Docker +
+    Sysbox will be strongly isolated from the host by Sysbox (e.g., via the Linux
+    user-namespace).
+
+* Also, in order for the Sysbox-In-Docker container to operate properly, the
+following host bind-mounts are required:
+
+    - "/var/lib/docker"
+
+    - "/var/lib/sysbox"
+
+    - "/lib/modules/<kernel>"
+
+    - "/usr/src/linux-headers-<kernel>"
+
+
+## Execution
+
+The Sysbox-In-Docker environment can be created in a few simple steps:
+
+1) Pick the Linux distribution on which to run Sysbox by choosing one of the
+Makefile targets displayed below.
+
+    ```
+    $ make sysbox-in-docker
+    ...
+    Usage:
+    make sysbox-in-docker <distro-release>
+
+    help                       Show supported docker images
+    centos-8                   Build CentOS-8 docker image
+    debian-buster              Build Debian-Buster docker image
+    debian-bullseye            Build Debian-Bullseye docker image
+    fedora-31                  Build Fedora-31 docker image
+    fedora-32                  Build Fedora-32 docker image
+    ubuntu-bionic              Build Ubuntu-Bionic docker image
+    ubuntu-focal               Build Ubuntu-Focal docker image
+    $
+    ```
+
+2) Build the Sysbox-In-Docker image by executing the chosen target. Notice that
+a sample instruction to launch the Sysbox-In-Docker container will be displayed
+as part of this output.
+
+    ```
+    $ make sysbox-in-docker ubuntu-bionic
+    ...
+    *** Launch container with the following instruction ***
+
+    docker run -it --privileged --rm --hostname sysbox-in-docker --name sysbox-in-docker -v /var/tmp/sysbox-var-lib-docker:/var/lib/docker -v /var/tmp/sysbox-var-lib-sysbox:/var/lib/sysbox -v /lib/modules/5.4.0-48-generic:/lib/modules/5.4.0-48-generic:ro -v /usr/src/linux-headers-5.4.0-48-generic:/usr/src/linux-headers-5.4.0-48-generic:ro -v /usr/src/linux-headers-5.4.0-48:/usr/src/linux-headers-5.4.0-48:ro nestybox/sysbox-in-docker:ubuntu-bionic
+
+    make: Nothing to be done for 'ubuntu-bionic'.
+    $
+    ```
+
+3) Create a Sysbox-In-Docker container by making use of the instruction obtained
+in the previous step:
+
+    ```
+    $ docker run -it --privileged --rm --hostname sysbox-in-docker --name sysbox-in-docker -v /var/tmp/sysbox-var-lib-docker:/var/lib/docker -v /var/tmp/sysbox-var-lib-sysbox:/var/lib/sysbox -v /lib/modules/5.4.0-48-generic:/lib/modules/5.4.0-48-generic:ro -v /usr/src/linux-headers-5.4.0-48-generic:/usr/src/linux-headers-5.4.0-48-generic:ro -v /usr/src/linux-headers-5.4.0-48:/usr/src/linux-headers-5.4.0-48:ro nestybox/sysbox-in-docker:ubuntu-bionic
+    ...
+    ```
+
+    In a separate terminal do ...
```

Comment:
Don't need this, since the `docker run` instruction uses `-it`.

Reply:
No, it's required if user doesn't specify an entrypoint, which is the case of the example above; IOW, by itself s6 won't run bash (or alike) to provide user a prompt. What you probably mean is to have the user doing "docker run -it <blah> bash", but that forces s6 to execute bash in the background which shows up like this in the container (see those 'ugly' foreground entries).

```
root@sysbox-in-docker:/# ps -ef
UID          PID    PPID  C STIME TTY          TIME CMD
root           1       0  0 07:08 pts/0    00:00:00 s6-svscan -t0 /var/run/s6/services
root          29       1  0 07:08 pts/0    00:00:00 foreground  if   /etc/s6/init/init-stage2-redirfd   foreground    if     if      s6-echo      -n      --      [s6-init] making user provided files available at /var/run/s6/etc...          foregr
root          30       1  0 07:08 pts/0    00:00:00 s6-supervise s6-fdholderd
root          37      29  0 07:08 pts/0    00:00:00 foreground  s6-setsid  -gq  --  with-contenv  backtick  -D  0  -n  S6_LOGGING   printcontenv   S6_LOGGING    importas  S6_LOGGING  S6_LOGGING  ifelse   s6-test   ${S6_LOGGING}   -eq   2     redi
root         223       1  0 07:08 pts/0    00:00:00 s6-supervise sysbox-fs
```

I don't mind changing the user instruction to match this approach, if you strongly prefer it let me know, just thought it would look cleaner like this.

Reply:
Got it. Sounds like we should replace `-it` with `-d` in the `docker run` instruction then.

Reply:
As mentioned, "-d" doesn't work with the existing s6 config. Will look into that and update dockerfiles once i find a solution.